### PR TITLE
[iOS Editor] Add support for saving live photo

### PR DESCRIPTION
**Pull request by ua741**
Note: This method only works when both image and video were part of the same live photo asset.

Related: #805

### CODE REVIEW

File: ios/Classes/PMPlugin.m [hunk 1]
```diff
@@ -376,6 +376,22 @@ - (void)onAuth:(FlutterMethodCall *)call result:(FlutterResult)result {
...
+        } else if ([call.method isEqualToString:@"saveLivePhoto"]) {
+            NSString *videoPath = call.arguments[@"videoPath"];
+            NSString *imagePath = call.arguments[@"imagePath"];
+            NSString *title = call.arguments[@"title"];
+            NSString *desc = call.arguments[@"desc"];
+            [manager saveLivePhoto:imagePath
+                         videoPath: videoPath
+                         title:title
+                         desc:desc
+                         block:^(PMAssetEntity *asset) {
```
Comment (AlexV525) [inline ios/Classes/PMPlugin.m:388]: Arguments seem not well-aligned here.

File: ios/Classes/core/PMManager.h [hunk 2]
```diff
@@ -65,6 +65,12 @@ typedef void (^AssetResult)(PMAssetEntity *);
              desc:(NSString *)desc
             block:(AssetResult)block;
 
+- (void)saveLivePhoto:(NSString *)imagePath
+            videoPath:(NSString *)videoPath
+            title:(NSString *)title
+             desc:(NSString *)desc
+            block:(AssetResult)block;
```
Comment (AlexV525) [inline ios/Classes/core/PMManager.h:72]: Same as above.

File: ios/Classes/core/PMManager.m [hunk 3]
```diff
@@ -1092,6 +1092,36 @@ - (void)saveVideo:(NSString *)path
     }];
 }
 
+- (void)saveLivePhoto:(NSString *)imagePath
+            videoPath:(NSString *)videoPath
+            title:(NSString *)title
+             desc:(NSString *)desc
+            block:(AssetResult)block {
```
Comment (AlexV525) [inline ios/Classes/core/PMManager.m:1099]: Same as above.

File: ios/Classes/core/PMManager.m [hunk 4]
```diff
@@ -1092,6 +1092,36 @@ - (void)saveVideo:(NSString *)path
...
+    [[PHPhotoLibrary sharedPhotoLibrary]
+     performChanges:^{
+        PHAssetCreationRequest *request = [PHAssetCreationRequest creationRequestForAsset];
+        PHAssetResourceCreationOptions *options = [PHAssetResourceCreationOptions new];
+        [options setOriginalFilename:title];
+        [request addResourceWithType:PHAssetResourceTypePhoto fileURL:imageURL options:options];
+        [request addResourceWithType:PHAssetResourceTypePairedVideo fileURL:fileURL options:options];
+        assetId = request.placeholderForCreatedAsset.localIdentifier;
+    }
+     completionHandler:^(BOOL success, NSError *error) {
```
Comment (AlexV525) [inline ios/Classes/core/PMManager.m:1114]: Did you make sure that the corresponding asset has a valid subtype to indicate as a live-photo?

File: lib/src/internal/editor.dart [hunk 5]
```diff
@@ -210,6 +210,29 @@ class IosEditor {
     }
     return null;
   }
+
+  /// Save Live photo to gallery from the given [imageFile] & [videoFile].
```
Comment (AlexV525) [inline lib/src/internal/editor.dart:214]:
```suggestion
  /// Save Live Photo to the gallery from the given [imageFile] and [videoFile].
```

File: lib/src/internal/plugin.dart [hunk 6]
```diff
@@ -331,6 +331,40 @@ class PhotoManagerPlugin with BasePlugin, IosPlugin, AndroidPlugin {
...
 
+  Future<AssetEntity?> saveLivePhoto({
+    required File imageFile,
+    required File videoFile,
+    required String? title,
+    String? desc,
+    String? relativePath,
+  }) async {
+    if (!imageFile.existsSync()) {
+      assert(false, 'imageFile must exists.');
```
Comment (AlexV525) [inline lib/src/internal/plugin.dart:342]:
```suggestion
      assert(false, 'File of the image must exist.');
```

File: lib/src/internal/plugin.dart [hunk 7]
```diff
@@ -331,6 +331,40 @@ class PhotoManagerPlugin with BasePlugin, IosPlugin, AndroidPlugin {
...
+      return null;
+    }
+    if (!videoFile.existsSync()) {
+      assert(false, 'videoFile must exists.');
```
Comment (AlexV525) [inline lib/src/internal/plugin.dart:346]:
```suggestion
      assert(false, 'File of the video must exist.');
```

**Review by AlexV525 — COMMENTED** [review]
Thanks for the contribution. I left some comments. Also, we would need an example to show how we can save live photos for iOS-only.

File: ios/Classes/core/PMManager.m (hunk 4 above)
Comment (ua741) [reply ios/Classes/core/PMManager.m:1114]: Yes, the native apple photos app also detected it as Live Photo.

File: README.md [hunk 8]
```diff
@@ -318,6 +318,16 @@ final AssetEntity? videoEntity = await PhotoManager.editor.saveVideo(
   videoFile, // You can check whether the file is exist for better test coverage.
   title: 'write_your_own_title.mp4',
 );
+
+// [iOS only] Save a live photo from image and video `File`.
+// This only works when both image and video file were part of same live photo.
+final File imageFile = File('path/to/your/livephoto.heic');
+final File videoFile = File('path/to/your/livevideo.mp4');
+final AssetEntity? entity = await PhotoManager.editor.saveLivePhoto(
```
Comment (AlexV525) [inline README.md:326]: You'll need to add this in the example project too.

Comment (ua741) [reply README.md:326]: Thank you @CaiJingLong for adding the example. Appreciate it! 
I was pondering over where I should upload the sample files :)

**Review by AlexV525 — APPROVED** [review]

**Pull request merged**